product_supplierinfo_discount/tests/test_product_supplierinfo_discount.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
+#        Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo.tests.common import TransactionCase
+from odoo import fields
+
+
+class TestProductSupplierinfoDiscount(TransactionCase):
+
+    def setUp(self):
+        super(TestProductSupplierinfoDiscount, self).setUp()
+        self.supplierinfo_model = self.env['product.supplierinfo']
+        self.purchase_order_line_model = self.env['purchase.order.line']
+        self.partner_1 = self.env.ref('base.res_partner_1')
+        self.partner_3 = self.env.ref('base.res_partner_3')
+        self.product = self.env.ref('product.product_product_6')
+        self.supplierinfo = self.supplierinfo_model.create({
+            'min_qty': 0.0,
+            'name': self.partner_3.id,
+            'product_tmpl_id': self.product.product_tmpl_id.id,
+            'discount': 10,
+        })
+        self.supplierinfo2 = self.supplierinfo_model.create({
+            'min_qty': 10.0,
+            'name': self.partner_3.id,
+            'product_tmpl_id': self.product.product_tmpl_id.id,
+            'discount': 20,
+        })
+        self.purchase_order = self.env['purchase.order'].create(
+            {'partner_id': self.partner_3.id})
+        self.po_line_1 = self.purchase_order_line_model.create(
+            {'order_id': self.purchase_order.id,
+             'product_id': self.product.id,
+             'date_planned': fields.Datetime.now(),
+             'name': 'Test',
+             'product_qty': 1.0,
+             'product_uom': self.env.ref('product.product_uom_categ_unit').id,
```

Comment:
@cubells IMO all `ref` should using in variable Because some `ref` are used many time.

here we using like different variable, @pedrobaeza what your opinion.